cloud/azure/actuators/machine/machine_status.go

```diff
@@ -0,0 +1,121 @@
+package machine
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/platform9/azure-provider/cloud/azure/services/resourcemanagement"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/util"
+)
+
+// holds the machine status under an annotation.
+// TODO: implement MachineStatus once the API is stable
```

Comment:
Main purpose is to store the previous machine status (in case of performing an update on the existing spec). This should be part of the `ProviderStatus` once the API is stable.

It also handles two other annotations for the machine's resource group and the name of the associate azure virtual machine.